examples/ecs/per_entity_events.rs

```diff
@@ -0,0 +1,283 @@
+    mut query: Query<Entity, With<Selectable>>,
+    mut selected: ResMut<Selected>,
+    keyboard_input: Res<Input<KeyCode>>,
+) {
+    let cycle_behavior: CycleBehavior = if keyboard_input.just_pressed(KeyCode::Tab) {
+        if keyboard_input.pressed(KeyCode::LShift) || keyboard_input.pressed(KeyCode::RShift) {
+            CycleBehavior::Back
+        } else {
+            CycleBehavior::Forward
+        }
+    } else {
+        return;
+    };
+
+    let mut entities = Vec::<Entity>::new();
+    // FIXME: Move to `.for_each` when https://github.com/bevyengine/bevy/issues/753 is resolved
+    query.for_each_mut(|entity| entities.push(entity.clone()));
+
+    let current_position = entities.iter().position(|&e| e == selected.entity).unwrap() as isize;
+
+    let new_position = match cycle_behavior {
+        // We have to convert to isize for this step to avoid underflows when current_postion == 0
+        CycleBehavior::Forward => (current_position + 1).rem_euclid(entities.len() as isize),
+        CycleBehavior::Back => (current_position - 1).rem_euclid(entities.len() as isize),
+    };
+
+    selected.entity = entities[new_position as usize];
+}
+
+fn scale_selected(
+    mut query: Query<(Entity, &mut Text), With<Selectable>>,
+    selected: Res<Selected>,
+) {
+    // Only do work when the selection is changed
+    if !selected.is_changed() {
+        return;
+    }
+
+    for (entity, mut text) in query.iter_mut() {
+        if entity == selected.entity {
+            text.sections[0].style.font_size = 90.0;
+        } else {
+            text.sections[0].style.font_size = 60.0;
+        }
+    }
+}
+
+// FIXME: make this work using `EventWriter<T>` syntax and specialized behavior
+/// Dispatches actions to entities based on the input
+/// Note that we can store several events at once!
+/// Try pressing both "1" and "3" to add 4 to the selected display
+fn input_dispatch(
+    // You could also access the &Events<T> component directly
+    // then send events to that component with `Events::send`
+    mut query: Query<
+        (&mut Events<CycleColorAction>, &mut Events<AddNumberAction>),
+        With<Selectable>,
+    >,
+    selected: Res<Selected>,
+    keyboard_input: ResMut<Input<KeyCode>>,
+) {
+    use KeyCode::*;
+
+    let (mut cycle_actions, mut add_actions) = query.get_mut(selected.entity).unwrap();
+
+    // Inputs for cycling colors
+    if keyboard_input.just_pressed(Space) {
+        cycle_actions.send(CycleColorAction);
+    }
+
+    // Inputs for sending numbers to be added
+    for k in keyboard_input.get_just_pressed() {
+        let num = match key_code {
+            Key0 => Some(0),
+            Key1 => Some(1),
+            Key2 => Some(2),
+            Key3 => Some(3),
+            Key4 => Some(4),
+            Key5 => Some(5),
+            Key6 => Some(6),
+            Key7 => Some(7),
+            Key8 => Some(8),
+            Key9 => Some(9),
+            _ => None,
+        };
+
+        if num.is_some() {
```

Comment:
I suspect this is not the simplest / most idiomatic way to handle this Option. Suggestions welcome.

Reply:
depends how much shortcut you want to take but I think `KeyCode::Key1 as u8` is 0 and they are in order with `KeyCode::Key1 as u8` being 10